tests/test_service.py

```diff
@@ -0,0 +1,78 @@
+import logging
+
+import pytest
+from conftest import proxy
+from plumbum import ProcessExecutionError
+from plumbum.cmd import docker
+
+logger = logging.getLogger()
+
+
+def _check_permissions(allowed_calls, forbidden_calls):
+    for args in allowed_calls:
+        docker(*args)
+    for args in forbidden_calls:
+        with pytest.raises(ProcessExecutionError):
+            docker(*args)
+
+
+def test_default_permissions():
+    with proxy() as test_container:
+        allowed_calls = (("version",),)
+        forbidden_calls = (
+            ("pull", "alpine"),
+            ("--rm", "alpine", "--name", test_container),
+            ("logs", test_container),
+            ("wait", test_container),
+            ("rm", "-f", test_container),
+            ("restart", test_container),
+            ("network", "ls"),
+            ("config", "ls"),
+            ("service", "ls"),
+            ("stack", "ls"),
+            ("secret", "ls"),
+            ("plugin", "ls"),
+            ("info",),
+            ("system", "info"),
+            ("build", "."),
+            ("swarm", "init"),
+        )
+        _check_permissions(allowed_calls, forbidden_calls)
```

Comment:
AFAICS all these tests can be joined in a single one which is paramtrized. But don't worry too much about it for now, they're also OK as they are.

Reply:
Ok, I ended up leaving as is to save time. Maybe we can leave it to a future improvement. WDYT?